docs/language-concepts.md

```diff
@@ -0,0 +1,450 @@
+# Language concepts
```

Comment:
I would call this Externals or FFI

Reply:
But the content of this page is a bit broader than just external and FFI, such as pipe operators